File: back/tp/funcional/Ac-3/src/Producto.java
public class Producto {
//roducto(nombre, categoria, precio, stock):
    private String nombre;
    private Double precio;
    private String categoria;
    private int stock;
    public Producto(String nombre,String categoria, Double precio, int stock) {
        this.nombre = nombre;
        this.precio = precio;
        this.stock = stock;
        this.categoria = categoria;
    }
    public String getNombre() {
        return nombre;
    }
    public Double getPrecio() {
        return precio;
    }
    public int getStock() {
        return stock;
    }
    public String getCategoria() {
        return categoria;
    }
}

File: bd/parcial/2/api-e-commerce/src/models/producto.js
import mongoose from "mongoose";
import { Schema } from "mongoose"

const productoSchema = mongoose.Schema({
    nombre:{type:String, require:true},
    descripcion:{type:String, require:true},
    categoriaId:{ type: Schema.Types.ObjectId, ref:'categoria', require:true},
    precio:{type:Number, require:true, min:0},
    stock:{type:Number, require:true, min:0}
})

export const Producto = mongoose.model('producto', productoSchema)

